### PR TITLE
Display tool source in tool form

### PR DESCRIPTION
**Pull request by mvdbeek**
Screenshot 2021-10-29 at 19 36 40

Screenshot 2021-10-29 at 19 36 22

TODO: 
- [x] create a standalone component
- [x] return tool source type in header
- [x] detect language
- [x] pass in tool version
- [x] tests
- [x] run some kind of prettify library over raw response
 
## How to test the changes?
(Select all options that apply)
- [ ] I've included appropriate [automated tests](https://docs.galaxyproject.org/en/latest/dev/writing_tests.html).
- [ ] This is a refactoring of components with existing test coverage.
- [ ] Instructions for manual testing are as follows:
  1. [add testing steps and prerequisites here if you didn't write automated tests covering all your changes]

## License
- [x] I agree to license these contributions under [Galaxy's current license](https://github.com/galaxyproject/galaxy/blob/dev/LICENSE.txt).
- [x] I agree to allow the Galaxy committers to license these and all my past contributions to the core galaxy codebase under the [MIT license](https://opensource.org/licenses/MIT). If this condition is an issue, uncheck and just let us know why with an e-mail to galaxy-committers@lists.galaxyproject.org.

### CODE REVIEW

**Comment by jmchilton** [comment]
Very cool. I think you should hide exposing this at the API level for non-admins behind a flag and then switch that flag to not hidden for usegalaxy.org.

File: client/src/components/Tool/ToolCard.vue [hunk 1]
```diff
@@ -26,6 +26,9 @@
                         <b-dropdown-item v-if="showDownload" @click="onDownload"
                             ><span class="fa fa-download" /><span v-localize>Download</span>
                         </b-dropdown-item>
+                        <b-dropdown-item v-b-modal.tool-source-viewer
+                            ><span class="fa fa-eye" /><span v-localize>Preview tool source</span>
```
Comment (guerler) [inline client/src/components/Tool/ToolCard.vue]: How about `View Tool source` or `View Tool XML`  in order to align it with `View Tool ID`?

File: lib/galaxy/webapps/galaxy/config_schema.yml [hunk 2]
```diff
@@ -2821,6 +2821,17 @@ mapping:
           Details" option in the history. This option also exposes the command line to
           non-administrative users. Administrators can always see dataset paths.
 
+      allow_tool_source_display:
```
Comment (bgruening) [inline lib/galaxy/webapps/galaxy/config_schema.yml]: Maybe `enable_tool_source_display` to be consistent with other options (enable_tool_recommendations, enable_markdown_export, enable_tool_tags)? Seems that enable is used more?

**Review by dannon — APPROVED** [review]
@mvdbeek Thanks for the fixup, looks great!

**Pull request merged**